added operationId to metadata of request specs

jamerine opened the pull request:
Added a change to allow users to add operationId to the metadata of request specs. There have been requests to make the  OpenAPI spec require operationId for many reasons. Here is an example: https://github.com/OAI/OpenAPI-Specification/issues/1907. 

If a user adds `operation-id` to the metadata of a request spec, it will be added to the path section of the schema.  

First time contributor and I struggled to figure out where to add a test for this feature. Let me know if there is a place where you would like to see a spec for this.

jamerine commented:
@exoego - What are your thoughts on this feature to enable `operationId` to be passed in as metadata options on a spec?

exoego commented:
Supporting `operation-id` itself sounds legit to me.
It is unclear to me that `path_server_records` or other props are added.

jamerine commented:
> Supporting `operation-id` itself sounds legit to me. It is unclear to me that `path_server_records` or other props are added.

@exoego - Sorry for the confusion here, I was testing out another proposed feature to the gem and committed my change to this PR. I am going to revert my last commit here so that this PR is strictly for the `operation-id` feature. Then I'll submit another PR for the server record changes.

jamerine commented:
@exoego - Ok, this PR for adding `operation-id` is good to review again.

exoego reviewed (commented):
@jamerine Could you update any single test to use `:operation_id` so we can ensure  OpenAPI file generated by tests have `operationId`.

jamerine commented:
@exoego - We added `:operation_id` to  a test and updated the expected.yaml to have the operationId as well. Let me know if there are any other changes you would like us to make.

jamerine commented:
@exoego - If we can get this `:operation_id` PR merged in, I will open up another PR for another proposed new feature.

exoego commented:
Sorry for the late response.
Thanks.
